src/main/java/moze_intel/projecte/gameObjs/items/rings/Arcana.java

```diff
@@ -1,155 +1,280 @@
 		
-		if (player.capabilities.isCreativeMode)
-		{
-			return;
-		}
+		tick(stack, world, (EntityPlayerMP)entity);
+	}
+
+	@Override
+	@Optional.Method(modid = "Baubles")
+	public BaubleType getBaubleType(ItemStack stack)
+	{
+		return BaubleType.RING;
+	}
+
+	@Override
+	@Optional.Method(modid = "Baubles")
+	public void onWornTick(ItemStack stack, EntityLivingBase entity)
+	{
+		if(stack.stackTagCompound == null) stack.setTagCompound(new NBTTagCompound());
 		
-		if (!playerMP.capabilities.allowFlying)
-		{
-			Utils.updateClientFlight(playerMP, true);
-			PlayerChecksEvent.addPlayerFlyChecks(playerMP);
-		}
+		if(entity.worldObj.isRemote || !(entity instanceof EntityPlayerMP)) return;
+		
+		tick(stack, entity.worldObj, (EntityPlayerMP)entity);
+	}
+
+	@Override
+	@Optional.Method(modid = "Baubles")
+	public void onEquipped(ItemStack stack, EntityLivingBase player)
+	{
 		
-		if (!player.isImmuneToFire())
+	}
+
+	@Override
+	@Optional.Method(modid = "Baubles")
+	public void onUnequipped(ItemStack stack, EntityLivingBase player)
+	{
+		
+	}
+
+	@Override
+	@Optional.Method(modid = "Baubles")
+	public boolean canEquip(ItemStack stack, EntityLivingBase player)
+	{
+		return true;
+	}
+
+	@Override
+	@Optional.Method(modid = "Baubles")
+	public boolean canUnequip(ItemStack stack, EntityLivingBase player)
+	{
+		return true;
+	}
+
+	@Override
+	public IIcon getIcon(ItemStack stack, int pass)
+	{
+		return getIconIndex(stack);
+	}
+
+	@Override
+	public IIcon getIconIndex(ItemStack stack)
+	{
+		boolean active = (stack.hasTagCompound() ? stack.getTagCompound().getBoolean("Active") : false);
+		return (active ? iconsOn : icons)[MathHelper.clamp_int(stack.getItemDamage(), 0, 3)];
+	}
+
+	@Override
+	public void registerIcons(IIconRegister register)
+	{
+		for(int i = 0; i < 4; i++)
 		{
-			Utils.setPlayerFireImmunity(player, true);
-			PlayerChecksEvent.addPlayerFireChecks((EntityPlayerMP) player);
+			icons[i] = register.registerIcon(this.getTexture("rings", "arcana_" + i));
 		}
 		
-		switch (stack.getItemDamage())
+		for(int i = 0; i < 4; i++)
 		{
-			case 0:
-				Utils.freezeNearby(world, entity);
-				break;
-			case 1:
-				Utils.igniteNearby(world, entity);
-				break;
-			case 2:
-				Utils.growNearbyRandomly(true, world, entity);
-				break;
-			case 3:
-				Utils.repelEntities(player);
-				break;
+			iconsOn[i] = register.registerIcon(this.getTexture("rings", "arcana_" + i + "_on"));
 		}
+		
+		itemIcon = icons[0];
 	}
-	
+
 	@Override
-	public ItemStack onItemRightClick(ItemStack stack, World world, EntityPlayer player)
+	public void addInformation(ItemStack stack, EntityPlayer player, List list, boolean b)
 	{
-		if (!world.isRemote)
+		if(stack.hasTagCompound())
 		{
-			if (stack.stackTagCompound.getBoolean("Active"))
+			if(!stack.stackTagCompound.getBoolean("Active"))
 			{
-				stack.stackTagCompound.setBoolean("Active", false);
+				list.add(EnumChatFormatting.RED + "Inactive!");
```

Comment:
I missed this the first time, but no raw English strings in the code. Do ``` + StatCollector.translateToLocal("pe.arcana.inactive");``` and then put ```pe.arcana.inactive=Inactive!``` in the en_US.lang. So it lets the string be translated to other languages